### PR TITLE
Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by argvniyx-enroute**
### Summary
This updates `test_track` to Rails 7 and fixes a breaking deprecation and a lint.

### Other Information
I'll comment on interesting stuff inline

/domain

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -52,13 +52,17 @@ gem 'foreman'
 gem 'delayed_job'
 gem 'delayed_job_active_record'
 
-gem 'with_transactional_lock'
+source "https://rubygems.pkg.github.com/betterment" do
+  gem "with_transactional_lock"
+end
```
[the remark this answers is not in the document]
Comment (smudge) [reply Gemfile:57]:
```suggestion
gem "with_transactional_lock"
```

File: Gemfile [hunk 2]
```diff
@@ -59,6 +59,8 @@ gem 'bootsnap', '>= 1.3.0', require: false
 # Avoid deprecation notices by pinning mail to Ruby 2.7.5 compatible version
 gem 'mail', '~> 2.7.1'
 
+gem 'sprockets-rails'
```
Comment (argvniyx-enroute) [inline Gemfile:62]: As y'all are aware, https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#sprockets-is-now-an-optional-dependency

**Review by smudge — APPROVED** [review]
domain LGTM && platform LGTM

File: config/initializers/types.rb [hunk 3]
```diff
@@ -1 +1,3 @@
-ActiveRecord::Type.register(:app_version, AppVersionType)
+Rails.application.config.to_prepare do
```
Comment (argvniyx-enroute) [inline config/initializers/types.rb:1]: This change comes from https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#autoloading-during-initialization. A couple of alternatives are listed in https://guides.rubyonrails.org/v7.0/autoloading_and_reloading_constants.html#autoloading-when-the-application-boots, but I'm not sure if any of the other use cases are a better fit.

**Pull request merged**